doc/README.md

```diff
@@ -0,0 +1,36 @@
+# Overview
+
+The `dart_dev` package includes:
+
+- Tools for running developer tasks via the `dart_dev` executable.
+- Utils for composing tools together.
+- Utils for creating your own tools.
+- A shared config that serves as a recommended starting place for your
+  `tool/dart_dev/config.dart`.
+
+---
+---
+
+<!-- Table of Contents -->
```

Comment:
I don't see a number of the markdown files linked to in this TOC (the same will be true of TOCs in some of the other docs as well as inline links in those docs to these tools). Are we planning to continue to flesh out this documentation after this PR merges?

Reply:
Ah, yeah I was originally hoping that I'd have those done in time for 3.0.0. I will remove them for now and we can add them back as they are implemented

Reply:
I also noticed some new features in dartdoc around generating docs in a more readme.io-esque way, so I'm going to look into that as a follow up. I still feel like there's value in these expanded guides being separate from the source docs, but it'd be nice to consolidate them into a single location if possible.

Reply:
(and presumably the table of contents would be auto-generated via dartdoc instead of having to do these manual updates)